### PR TITLE
Response Caching Middleware 2.2 update

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #5495

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/performance/caching/middleware?view=aspnetcore-2.1&branch=pr-en-us-13132)

**No rush on review.**

### CODE REVIEW

File: aspnetcore/performance/caching/middleware.md [hunk 1]
```diff
@@ -53,35 +42,36 @@ Response Caching Middleware only caches server responses that result in a 200 (O
...
-    options.UseCaseSensitivePaths = true;
     options.MaximumBodySize = 1024;
+    options.UseCaseSensitivePaths = true;
 });
 ```
 
 ## VaryByQueryKeys
 
-When using MVC/Web API controllers or Razor Pages page models, the `ResponseCache` attribute specifies the parameters necessary for setting the appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is `VaryByQueryKeys`, which doesn't correspond to an actual HTTP header. For more information, see [ResponseCache Attribute](xref:performance/caching/response#responsecache-attribute).
+When using MVC/Web API controllers or Razor Pages page models, the <xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute> attribute specifies the parameters necessary for setting the appropriate headers for response caching. The only parameter of the `ResponseCache` attribute that strictly requires the middleware is <xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute.VaryByQueryKeys>, which doesn't correspond to an actual HTTP header. For more information, see <xref:performance/caching/response#responsecache-attribute>.
```
Comment (scottaddie) [inline aspnetcore/performance/caching/middleware.md]:
```suggestion
When using MVC / web API controllers or Razor Pages page models, the [[ResponseCache]](xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute) attribute specifies the parameters necessary for setting the appropriate headers for response caching. The only parameter of the `[ResponseCache]` attribute that strictly requires the middleware is <xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute.VaryByQueryKeys>, which doesn't correspond to an actual HTTP header. For more information, see <xref:performance/caching/response#responsecache-attribute>.
```

File: aspnetcore/performance/caching/middleware.md [hunk 2]
```diff
@@ -53,35 +42,36 @@ Response Caching Middleware only caches server responses that result in a 200 (O
...
 
-When not using the `ResponseCache` attribute, response caching can be varied with the `VaryByQueryKeys` feature. Use the `ResponseCachingFeature` directly from the `IFeatureCollection` of the `HttpContext`:
+When not using the `ResponseCache` attribute, response caching can be varied with `VaryByQueryKeys`. Use the <xref:Microsoft.AspNetCore.ResponseCaching.ResponseCachingFeature> directly from the [HttpContext.Features](xref:Microsoft.AspNetCore.Http.HttpContext.Features):
```
Comment (scottaddie) [inline aspnetcore/performance/caching/middleware.md]:
```suggestion
When not using the `[ResponseCache]` attribute, response caching can be varied with `VaryByQueryKeys`. Use the <xref:Microsoft.AspNetCore.ResponseCaching.ResponseCachingFeature> directly from the [HttpContext.Features](xref:Microsoft.AspNetCore.Http.HttpContext.Features):
```

File: aspnetcore/performance/caching/middleware.md [hunk 3]
```diff
@@ -2,49 +2,38 @@
...
 
 By [Luke Latham](https://github.com/guardrex) and [John Luo](https://github.com/JunTaoLuo)
 
-[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/performance/caching/middleware/samples) ([how to download](xref:index#how-to-download-a-sample)).
+[View or download sample code](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/performance/caching/middleware/samples) ([how to download](xref:index#how-to-download-a-sample))
 
-This article explains how to configure Response Caching Middleware in an ASP.NET Core app. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the `ResponseCache` attribute, see [Response Caching](xref:performance/caching/response).
+This article explains how to configure Response Caching Middleware in an ASP.NET Core app. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the <xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute> attribute, see [Response Caching](xref:performance/caching/response).
```
Comment (scottaddie) [inline aspnetcore/performance/caching/middleware.md]:
```suggestion
This article explains how to configure Response Caching Middleware in an ASP.NET Core app. The middleware determines when responses are cacheable, stores responses, and serves responses from cache. For an introduction to HTTP caching and the [[ResponseCache]](xref:Microsoft.AspNetCore.Mvc.ResponseCacheAttribute) attribute, see [Response Caching](xref:performance/caching/response).
```

File: aspnetcore/performance/caching/middleware.md [hunk 4]
```diff
@@ -2,49 +2,38 @@
...
-
-::: moniker range="= aspnetcore-2.0"
-
-Reference the [Microsoft.AspNetCore.All metapackage](xref:fundamentals/metapackage) or add a package reference to the [Microsoft.AspNetCore.ResponseCaching](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package.
-
-::: moniker-end
+## Configuration
 
-::: moniker range="= aspnetcore-1.1"
+Use the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) or add a package reference to the [Microsoft.AspNetCore.ResponseCaching](https://www.nuget.org/packages/Microsoft.AspNetCore.ResponseCaching/) package.
```
Comment (scottaddie) [inline aspnetcore/performance/caching/middleware.md]: The metapackage won't be used for 3.0+. Do you have another issue to address this?

Comment (guardrex) [reply aspnetcore/performance/caching/middleware.md]: Not addressing it on this little pass. I only have another ... er ... maybe four samps to hit for 2.2.

WRT the `App` package text changing, I'll take care of it for my 3.0 pass on all of my authored topics later. This pass is 2.2 only.

Comment (guardrex) [reply aspnetcore/performance/caching/middleware.md]: Issue is the same ... last column of #5495 ... I guessed we'd use the same process as has been used in prior releases.

However, guidance might be shifting: It seems like extra work to open one separate issue for every topic. I can just use #5495. **_But_...** if the team is judged on the **_number of issues_** resolved in a given period, then that's probably the right thing to do. I figured that we're far enough off from 3.0 at this time that I can get a final answer on that whenever. It doesn't seem critical at the moment.

**Pull request merged**